src/store/actions.ts

```diff
@@ -152,58 +199,30 @@ export default {
     }
   }),
   save: tryAction(async ({ state, commit }: {state: ApplicationState, commit: any}) => {
-    const body = { contents: JSON.stringify(toCart(state)) }
-    const response = await api.post('/api/v1/lifelines_cart', { body: JSON.stringify(body) })
-    const location: string = response.headers.get('Location')
-    const id: string = location.substring(location.lastIndexOf('/') + 1)
-    commit('setToast', { type: 'success', message: 'Saved order with id ' + id })
-    router.push({ name: 'load', params: { cartId: id } })
+    const formFields = [...state.orderFormFields, { id: 'contents', type: 'text' }]
+    const formData = { ...state.order, ...{ contents: JSON.stringify(toCart(state)) } }
+
+    if (state.order.orderNumber) {
+      await updateOrder(formData, formFields)
+      commit('setToast', { type: 'success', message: 'Saved order with order number ' + state.order.orderNumber })
+      return state.order.orderNumber
+    } else {
+      const orderNumber = await createOrder(formData, formFields).catch(() => {
+        return Promise.reject(new Error('Failed to create order'))
+      })
+      const newOrderResponse = await api.get(`/api/v2/lifelines_order/${orderNumber}`)
+      commit('restoreOrderState', newOrderResponse)
+      commit('setToast', { type: 'success', message: 'Saved order with order number ' + orderNumber })
```

Comment:
Maybe differentiate between order submission and saving the order?

Reply:
renamed the button to save , will add the submit ( including state change) in later PR